x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -616,25 +647,24 @@ export class TaskRunner<
       ...(namespace ? { namespace } : {}),
     });
 
-    this.alertingEventLogger.start();
+    this.alertingEventLogger.start(this.runDate);
 
-    return await loadRule<Params>({
+    return validateRule({
+      alertingEventLogger: this.alertingEventLogger,
+      ruleData: this.ruleData!,
```

Comment:
Especially deep down in the bowels of the rule runner, I don't think we should be using `!` to force the value.  I'd prefer we check it before this call and throw an error that we can trace back to this code; "ruleData not available when preparing to run rule" or similar.

Reply:
Done